tests/e2e/metrics_test.go

```diff
@@ -56,18 +57,20 @@ var _ = Describe("Metrics", func() {
 	var err error
 	// We define a few metrics in the tests. We check that all of them exist and
 	// there are no errors during the collection.
-	metricsRegexp := regexp.MustCompile(
-		`(?m:^(` +
-			`cnpg_pg_postmaster_start_time_seconds \d+\.\d+|` + // wokeignore:rule=master
-			`cnpg_pg_wal_files_total \d+|` +
-			`cnpg_pg_database_size_bytes{datname="app"} [0-9e\+\.]+|` +
-			`cnpg_pg_replication_slots_inactive 0|` +
-			`cnpg_pg_stat_archiver_archived_count \d+|` +
-			`cnpg_pg_stat_archiver_failed_count \d+|` +
-			`cnpg_pg_locks_blocked_queries 0|` +
-			`cnpg_runonserver_match 42|` +
-			`cnpg_collector_last_collection_error 0)` +
-			`$)`)
+	metricsList := []interface{}{
+		"cnpg_pg_postmaster_start_time_seconds \\d+\\.\\d+|", // wokeignore:rule=master
+		"cnpg_pg_wal_files_total \\d+|",
+		"cnpg_pg_database_size_bytes{datname=\"app\"} [0-9e\\+\\.]+|",
+		"cnpg_pg_replication_slots_inactive 0|",
+		"cnpg_pg_stat_archiver_archived_count \\d+|",
+		"cnpg_pg_stat_archiver_failed_count \\d+|",
+		"cnpg_pg_locks_blocked_queries 0|",
+		"cnpg_runonserver_match_fixed 42|",
+		"cnpg_collector_last_collection_error 0)",
+	}
+
+	metricsRegexp := regexp.MustCompile(fmt.Sprintf(`(?m:^(`+`%v`+`%v`+`%v`+`%v`+`%v`+`%v`+`%v`+`%v`+`%v`+`$)`,
```

Comment:
Unfortunately this regex is poorly readable and dependent on the length of `metricsList`.
I'd like, if possible, to have something that's more generic.
Perhaps splitting the metrics output on lines and counting just the metrics we are interested in would be better?

Reply:
@leonardoce and @armru Fixed. 
Please review.